cpp/common/src/codingstandards/cpp/rules/memoryoperationsnotsequencedappropriately_shared/MemoryOperationsNotSequencedAppropriately_shared.qll

```diff
@@ -0,0 +1,251 @@
+/**
+ * Provides a library which includes a `problems` predicate for reporting....
+ */
+
+import cpp
+import codingstandards.cpp.Customizations
+import codingstandards.cpp.CExpr
+import codingstandards.cpp.Exclusions
+import codingstandards.cpp.SideEffects
+import codingstandards.cpp.COrdering
```

Comment:
If this shared query is being used for a C++ rule, is C ordering appropriate here?

Reply:
Re-reviewing the rule, I think I made a mistake in identifying this as an "Import" - sorry about that! While the sequencing rules are similar between C++ and C, they are not identical. Furthermore, C++17 introduced changes as noted by the rule:

> Note: C++17 changed the evaluation order of several expressions from indeterminately sequenced to
sequenced before, which means that code that is compliant with this rule may not work correctly with
earlier versions of C++.

@mbaluda I think we should revert the switch to a shared query, and move the `COrdering` library back to the `c/common` pack.

When we come to implement this rule by hand, we can certainly be influenced by the C rule - so I would leave it as a "similar" rule - but I don't think the implementation is going to be fully shared.